errors/ControllerErrorHandler.spec.ts

```diff
@@ -0,0 +1,91 @@
+import chai = require('chai');
+import sinon = require('sinon');
+import sinonChai = require('sinon-chai');
+import {ErrorHandlerWrongArgumentsError, RouteError} from './Errors';
+import {ControllerErrorHandler} from './ControllerErrorHandler';
+
+let should = chai.should();
+chai.use(sinonChai);
+
+describe('ControllerErrorHandler', () => {
+
+    let manager: ControllerErrorHandler;
+
+    beforeEach(() => {
+        manager = new ControllerErrorHandler();
+    });
+
+    it('should add a handler to default', () => {
+        let func = (req, res, err) => console.log(err);
+
+        manager.addHandler(func);
+
+        (manager as any).handlers.Error.should.equal(func);
+    });
+
+    it('should add a handler to a specific error', () => {
+        let func = (req, res, err) => console.log(err);
+
+        manager.addHandler(func, RouteError);
+
+        (manager as any).handlers.RouteError.should.equal(func);
+    });
+
+    it('should add a handler to multiple specific errors', () => {
+        let func = (req, res, err) => console.log(err);
+
+        manager.addHandler(func, RouteError);
+        manager.addHandler(func, ErrorHandlerWrongArgumentsError);
+
+        (manager as any).handlers.RouteError.should.equals(func);
+        (manager as any).handlers.ErrorHandlerWrongArgumentsError.should.equals(func);
+    });
+
+    it('should replace the handler if another is registered', () => {
+        let func = (req, res, err) => console.log(err);
+        let func2 = (req, res, err) => console.log(err);
+
+        manager.addHandler(func, RouteError);
+        manager.addHandler(func2, RouteError);
+
+        (manager as any).handlers.RouteError.should.equal(func2);
```

Comment:
Maybe add a console.warn if you register two errohandler for the same error type?